diff -up mesa-20170119/src/gallium/auxiliary/gallivm/lp_bld_gather.c.endian mesa-20170119/src/gallium/auxiliary/gallivm/lp_bld_gather.c
--- mesa-20170119/src/gallium/auxiliary/gallivm/lp_bld_gather.c.endian	2017-01-19 13:16:50.650522943 +1000
+++ mesa-20170119/src/gallium/auxiliary/gallivm/lp_bld_gather.c	2017-01-19 13:16:59.336753215 +1000
@@ -527,7 +527,7 @@ lp_build_gather(struct gallivm_state *ga
       if (vec_zext) {
          res = LLVMBuildZExt(gallivm->builder, res, res_t, "");
          if (vector_justify) {
-#if PIPE_ARCH_BIG_ENDIAN
+#ifdef PIPE_ARCH_BIG_ENDIAN
             unsigned sv = dst_type.width - src_width;
             res = LLVMBuildShl(gallivm->builder, res,
                                lp_build_const_int_vec(gallivm, res_type, sv), "");
